src/runtime/components/DocSearch.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div id="docsearch">
+    <button type="button" class="DocSearch DocSearch-Button justify-end" aria-label="Search">
+      <svg width="20" height="20" class="d-icon m-auto" viewBox="0 0 20 20">
+        <path
+          d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z"
+          stroke="currentColor"
+          fill="none"
+          fill-rule="evenodd"
+          stroke-linecap="round"
+          stroke-linejoin="round"
+        />
+      </svg>
+    </button>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { useRoute, useRouter } from '#imports'
+
+const route = useRoute()
+
+const router = useRouter()
+
+const props = defineProps({
+  options: {
+    type: Object,
+    required: true
+  }
+})
+
+const isSpecialClick = event => event.button === 1 || event.altKey || event.ctrlKey || event.metaKey || event.shiftKey
+
+const stripTrailingSlash = url => url.replace(/\/$|\/(?=\?)|\/(?=#)/g, '')
+
+const getRelativePath = (absoluteUrl) => {
+  const { pathname, hash } = new URL(absoluteUrl)
+  const url = pathname.replace('https://v3.nuxtjs.org', '/') + hash
+  return stripTrailingSlash(url)
+}
+
+const initialize = async (userOptions: any, code: string) => {
+  const lang = code || 'en-US'
+
+  const docsearch = await Promise.all([
+    // @ts-ignore
+    import(/* webpackChunkName: "docsearch" */ '@docsearch/js'),
+    // @ts-ignore
+    import(/* webpackChunkName: "docsearch" */ '@docsearch/css')
+  ]).then(([docsearch]) => docsearch.default)
+
+  // Create DocSearch instance
+  docsearch({
+    ...userOptions,
+    container: '#docsearch',
+    searchParameters: {
+      ...((!lang)
+        ? {}
+        : {
+            facetFilters: [`${userOptions.langAttribute || 'language'}:${lang}`].concat(
+              userOptions.facetFilters || []
```

Comment:
It would be useful to add types for userOptions/props.options so that we could have a type support here

Reply:
Updated it, you can take a look at AlgoliaDocSearch component!